src/Language/Pirouette/QuasiQuoter/Syntax.hs

```diff
@@ -114,23 +113,19 @@ data Expr lang
   | ExprLit (Constants lang)
   | ExprBase (BuiltinTerms lang)
   | ExprIf (Ty lang) (Expr lang) (Expr lang) (Expr lang)
+  | ExprBottom (Ty lang)
```

Comment:
Ok, I was reviewing this more carefully — why do we need bottoms again?

Reply:
Because otherwise we can't really write terms such as `fromJust` in a predicate, its definition would be:
```
fun fromJust : all (a : Type) . Maybe a -> a
  = /\ (a : Type) . \ (ma : Maybe a) . match_Maybe @a ma (\(x : a) -> a) (bottom @a)
```

This is useful because we might want to check the following predicate:
```
[term| (\(res : Result) (i : Maybe Integer) . res && isJust i) |] :==>: [term| (\res : Result) (i : Maybe Integer) . fromJust i > 10 |]
```

i.e., we know a property about the input in the antecedent of the implication, and we want to extract its value in the
consequent, but for that we need `fromJust`, which doesn't have a definition without `bottom`.